File: 2224-minimum-number-of-operations-to-convert-time/2224-minimum-number-of-operations-to-convert-time.java
class Solution {
    public int convertTime(String current, String correct) {
        String[] _correct = correct.split(":");
        int target = Integer.parseInt(_correct[0]) * 60 + Integer.parseInt(_correct[1]);
        String[] _current = current.split(":");
        int currTime = Integer.parseInt(_current[0]) * 60 + Integer.parseInt(_current[1]);
        int count = 0;
        while(target != currTime) {
            if (currTime + 60 <= target) currTime += 60;
            else if (currTime + 15 <= target) currTime += 15;
            else if (currTime + 5 <= target) currTime += 5;
            else if (currTime + 1 <= target) currTime += 1;
            count++;
        }
        return count;
    }
}

File: 2224-minimum-number-of-operations-to-convert-time/2224-minimum-number-of-operations-to-convert-time.kt
class Solution {
    fun convertTime(current: String, correct: String): Int {
        val minutes1 = current.split(":")[0].toInt() * 60 + current.split(":")[1].toInt()        
        val minutes2 = correct.split(":")[0].toInt() * 60 + correct.split(":")[1].toInt()
        
        val diff = minutes2 - minutes1
        
        if (diff == 0) return 0
        
        val hourModulo = diff % 60
        val quarterModulo = diff % 60 % 15
        
        var operations = diff / 60
        operations += hourModulo / 15
        operations += quarterModulo / 5
        operations += quarterModulo % 5
        
        return operations
    }
}